Cms/ProfileBioEdit.ascx.cs

```diff
@@ -0,0 +1,2394 @@
+                                familyGroup.SaveAttributeValues();
+                            }
+                        }
+                    }
+
+                    return returnVal;
+                } );
+
+                if ( wrapTransactionResult )
+                {
+                    // If there's a ReturnUrl specified navigate to that page.
+                    // Otherwise stay on the page, but show a saved success modal.
+                    var returnUrl = PageParameter( PageParameterKey.ReturnUrl );
+
+                    if ( returnUrl.IsNotNullOrWhiteSpace() )
+                    {
+                        string redirectUrl = Server.UrlDecode( returnUrl );
+
+                        string queryString = string.Empty;
+                        if ( redirectUrl.Contains( "?" ) )
+                        {
+                            queryString = redirectUrl.Split( '?' ).Last();
+                        }
+                        Context.Response.Redirect( redirectUrl );
+                    }
+                    else
+                    {
+                        NavigateToLinkedPage( AttributeKey.RedirectPage );
+                    }
+
+                    maAlert.Show( "Your profile has been updated!", ModalAlertType.None );
+                }
+            }
+        }
+
+        /// <summary>
+        /// Handles the Click event of the lbCancel control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
+        protected void lbCancel_Click( object sender, EventArgs e )
+        {
+            var redirect = NavigateToLinkedPage( AttributeKey.RedirectPage );
+            if ( !redirect )
+            {
+                NavigateToParentPage();
+            }
+        }
+
+        /// <summary>
+        /// Handles the Click event of the lbAddFamilyMember control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
+        protected void lbAddFamilyMember_Click( object sender, EventArgs e )
+        {
+            var pnlFamilyMemberBody = pnlProfilePanels.FindControl( "pnlFamilyMemberBody" );
+            if ( pnlFamilyMemberBody != null )
+            {
+                var lastPersonId = FamilyMembers.OrderByDescending( fm => fm.PersonId ).Select( fm => fm.PersonId ).LastOrDefault();
+                if ( lastPersonId > 0 )
+                {
+                    lastPersonId = 0;
+                }
+                lastPersonId--;
+
+                var newFamilyMember = new PersonFamilyMember
+                {
+                    Guid = Guid.Empty,
+                    PersonId = lastPersonId,
+                    Person = new Person()
+                };
+                newFamilyMember.Person.LastName = "New Family Member";
```

Comment:
Seeif this can be handled better by setting the widget Person object with PersonFamilyMember.Person